examples/chef/common/chef-rvc-mode-delegate.cpp

```diff
@@ -106,17 +108,61 @@ void RvcRunMode::Shutdown()
     }
 }
 
+chip::Protocols::InteractionModel::Status chefRvcRunModeWriteCallback(chip::EndpointId endpointId, chip::ClusterId clusterId,
+                                                                      const EmberAfAttributeMetadata * attributeMetadata,
+                                                                      uint8_t * buffer)
+{
+    chip::Protocols::InteractionModel::Status ret = Protocols::InteractionModel::Status::Success;
+
+    VerifyOrDie(endpointId == 1); // this cluster is only enabled for endpoint 1.
+    VerifyOrDie(gRvcRunModeInstance != nullptr);
+
+    ModeBase::Instance * clusterInstance = gRvcRunModeInstance;
+    chip::AttributeId attributeId        = attributeMetadata->attributeId;
+
+    switch (attributeId)
+    {
+    case chip::app::Clusters::RvcRunMode::Attributes::CurrentMode::Id: {
+        uint8_t m                                        = static_cast<uint8_t>(buffer[0]);
+        chip::Protocols::InteractionModel::Status status = clusterInstance->UpdateCurrentMode(m);
+        if (chip::Protocols::InteractionModel::Status::Success == status)
+        {
+            break;
+        }
+        ret = chip::Protocols::InteractionModel::Status::UnsupportedWrite;
```

Comment:
what kind of status may UpdateCurrentMode return that we convert always to UnsupportedWrite? Why are we not doing `ret = clusterInstance->UpdateCurrentMode(m)` ? a comment should explain

Reply:
good point. thanks. Updated